pkg/rancher-desktop/sudo-prompt/index.js

```diff
@@ -379,83 +325,48 @@ function MacCommand(instance, end) {
   Node.fs.writeFile(path, script, 'utf-8', end);
 }
 
-function MacIcon(instance, end) {
-  if (!instance.options.icns) {
-    return end();
-  }
-  Node.fs.readFile(instance.options.icns,
-    (error, buffer) => {
-      if (error) {
-        return end(error);
-      }
-      const icns = Node.path.join(
-        instance.path,
-        'Contents',
-        'Resources',
-        'applet.icns',
-      );
+function MacOpen(instance, end) {
+  let basePath;
 
-      Node.fs.writeFile(icns, buffer, end);
-    },
-  );
-}
+  if (Node.electron.app?.isPackaged) {
+    basePath = process.resourcesPath;
+  } else {
+    basePath = process.cwd();
+  }
 
-function MacOpen(instance, end) {
-  // We must run the binary directly so that the cwd will apply.
-  const binary = Node.path.join(instance.path, 'Contents', 'MacOS', 'applet');
-  // We must set the cwd so that the AppleScript can find the shell scripts.
+  // We must set the cwd so that the AppleScript can find the sudo-prompt-command script.
   const options = {
-    cwd:      Node.path.dirname(binary),
+    cwd:      instance.path,
     encoding: 'utf-8',
   };
 
-  // We use the relative path rather than the absolute path. The instance.path
-  // may contain spaces which the cwd can handle, but which exec() cannot.
-  Node.child.exec(`./${ Node.path.basename(binary) }`, options, end);
-}
+  if (Node.process.env.RD_SUDO_PROMPT_OSASCRIPT) {
+    const script = Node.path.join(basePath, 'resources', 'darwin', 'internal', 'Rancher Desktop.app', 'Contents', 'Resources', 'Scripts', 'main.scpt');
 
-function MacPropertyList(instance, end) {
-  // Value must be in single quotes (not double quotes) according to man entry.
-  // e.g. defaults write com.companyname.appname "Default Color" '(255, 0, 0)'
-  // The defaults command will be changed in an upcoming major release to only
-  // operate on preferences domains. General plist manipulation utilities will
-  // be folded into a different command-line program.
-  const plist = Node.path.join(instance.path, 'Contents', 'Info.plist');
-  const path = EscapeDoubleQuotes(plist);
-  const key = EscapeDoubleQuotes('CFBundleName');
-  const value = `${ instance.options.name } Password Prompt`;
-
-  if (/'/.test(value)) {
-    return end(new Error('Value should not contain single quotes.'));
-  }
-  let command = [];
+    Node.child.exec(`/usr/bin/osascript "${ EscapeDoubleQuotes(Node.path.normalize(script)) }"`, options, end);
```

Comment:
I'd rather we used `execFile` (so we don't need to escape args), but that can happen when we do a rewrite…